src/main/java/nwoolcan/controller/ProductionControllerImpl.java

```diff
@@ -0,0 +1,34 @@
+package nwoolcan.controller;
+
+import nwoolcan.model.brewery.Brewery;
+import nwoolcan.model.brewery.production.batch.QueryBatch;
+import nwoolcan.model.brewery.production.batch.QueryBatchBuilder;
+import nwoolcan.viewmodel.brewery.production.ProductionViewModel;
+import nwoolcan.viewmodel.brewery.production.batch.MasterBatchViewModel;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+//Package-private.
+final class ProductionControllerImpl implements ProductionController {
+
+    private final Brewery brewery;
```

Comment:
it is a singleton

Reply:
it is a first refactor for using less sinlgetons. for now i will just pick the brewery context for it but in the future i may not